src/poll/responser.rs

```diff
@@ -0,0 +1,666 @@
+// TODO: this `#![allow(unused)]` lint is temporarily added. Will be removed sooner.
+#![allow(unused)]
+use std::collections::HashMap;
+use std::sync::RwLock;
+use std::{net::Ipv4Addr, slice::from_raw_parts_mut, sync::Arc, thread::spawn};
+
+use bitfield::bitfield;
+use libc::IFA_ADDRESS;
+use lockfree::queue::Queue;
+
+use crate::qp::QpContext;
+use crate::{device::QpType::RawPacket, Sge};
+use crate::{
+    device::{
+        ToCardWorkRbDescBuilder, ToCardWorkRbDescCommon, ToHostWorkRbDescAethCode,
+        ToHostWorkRbDescOpcode, ToHostWorkRbDescRead,
+    },
+    Device, Error,
+};
+
+/// A single MR that contains a free list of packet buffer. Each slot is 64 Byte.
+pub(crate) const ACKNOWLEDGE_BUFFER_SLOT: usize = 64;
+/// The size of the ack buffer.
+pub(crate) const ACKNOWLEDGE_BUFFER_SIZE: usize = ACKNOWLEDGE_BUFFER_SLOT * 1024;
+
+/// A interface that allows DescResponser to push the work descriptor to the device
+pub trait WorkDescriptorSender: Send + Sync {
```

Comment:
A "Sender" trait  defined in a module under "poll" is confusing. Poll should be a kind of action related to Receive, not send

Reply:
Yes. The `responser.rs` is responsible for sending the response. I wonder if should I move this mod(`responser.rs`) outside the `poll ' directory?

Reply:
yes, try find a new place for them.